SQL/Archive/2017-11-01-use_consent_to_database.sql

```diff
@@ -0,0 +1,35 @@
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'useConsent', 'Enable if the study uses the loris architecture for consent', 1, 0, 'boolean', ID, 'Use consent', 15 FROM ConfigSettings WHERE Name='study';
+INSERT INTO Config (ConfigID, Value) SELECT ID, 'false' FROM ConfigSettings WHERE Name='useConsent';
+
+CREATE TABLE `consent_type` (
+  `ConsentTypeID` int(2) NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `Label` varchar(255) NOT NULL,
+  CONSTRAINT `PK_consent_type` PRIMARY KEY (`ConsentTypeID`),
+  CONSTRAINT `UK_consent_type_Name` UNIQUE KEY `Name` (`Name`),
+  CONSTRAINT `UK_consent_type_Label` UNIQUE KEY `Label` (`Label`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `candidate_consent_type_rel` (
+  `CandidateID` int(6) NOT NULL,
+  `ConsentTypeID` int(2) NOT NULL,
+  `Status` enum('yes', 'no', 'not_answered') NOT NULL,
+  `DateGiven` date DEFAULT NULL,
```

Comment:
as per https://github.com/aces/Loris/blob/19.0-dev/docs/SQLModelingStandard.md (Which should be propagated to minor),
ENUM are not permitted.

Could use this occasion to add a table "option" with "TableName", "AttributeName", "Value" attributes. This would create the table to replace all ENUM within the DB. Don't forget the (artificial) primary key.

Reply:
Im not sure about this design... I actually told zaliqa to keep it as an enum until we discuss how to tackle it.

Sever places will use the yes/no options in loris and im not sure if those options should also include "not answered" or other options.

Id like to bring this up at the loris meeting so it can be discussed with everyone

Reply:
yepp, let's put it through as enum for now until a standard for all options in the database is decided on

Reply:
Excellent idea, we could create a class for that so it be easy on projects to use the feature. I should work on that after I'm finish with visit

Reply:
the reason for yes/no enums in LORIS is that MySQL mutilates "boolean" to tinyint(1) and changes true/false to 1/0.

Maybe we should document that and update the SQL standards?  (not answered doesn't make sense though)

Reply:
yeah i agree, i wanted to talk about the not answered option at tomorrow's meeting